python/dgl/distributed/sampling.py

```diff
@@ -0,0 +1,104 @@
+"""Sampling module"""
+from .rpc import Request, Response, remote_call_to_machine
+from .dist_graph import DistGraph
+from ..sampling import sample_neighbors as local_sample_neighbors
+from . import register_service
+from ..convert import graph
+from ..base import NID, EID
+from .. import backend as F
+
+__all__ = ['sample_neighbors']
+
+SAMPLING_SERVICE_ID = 6657
+
+
+class SamplingResponse(Response):
+    """Sampling Response"""
+
+    def __init__(self, global_src, global_dst, global_eids):
+        self.global_src = global_src
+        self.global_dst = global_dst
+        self.global_eids = global_eids
+
+    def __setstate__(self, state):
+        self.global_src, self.global_dst, self.global_eids = state
+
+    def __getstate__(self):
+        return self.global_src, self.global_dst, self.global_eids
+
+
+class SamplingRequest(Request):
+    """Sampling Request"""
+
+    def __init__(self, nodes, fan_out, edge_dir='in', prob=None, replace=False):
+        self.seed_nodes = nodes
+        self.edge_dir = edge_dir
+        self.prob = prob
+        self.replace = replace
+        self.fan_out = fan_out
+
+    def __setstate__(self, state):
+        self.seed_nodes, self.edge_dir, self.prob, self.replace, self.fan_out = state
+
+    def __getstate__(self):
+        return self.seed_nodes, self.edge_dir, self.prob, self.replace, self.fan_out
+
+    def process_request(self, server_state):
+        local_g = server_state.graph
+        partition_book = server_state.partition_book
+        local_ids = F.astype(partition_book.nid2localnid(
+            F.tensor(self.seed_nodes), partition_book._part_id), local_g.idtype)
```

Comment:
Here I use a private attribute. I see it's exposed in PartitionPolicy. Do I need to add a new attribute to GraphPartitionBook & RangePartitionBook?

Reply:
yes, please don't use a private attribute. we can potentially keep the partition id in the server state.